extensions/azure-functions/runtime/src/main/java/io/quarkus/azure/functions/runtime/QuarkusAzureFunctionsMiddleware.java

```diff
@@ -0,0 +1,23 @@
+package io.quarkus.azure.functions.runtime;
+
+import com.microsoft.azure.functions.internal.spi.middleware.Middleware;
+import com.microsoft.azure.functions.internal.spi.middleware.MiddlewareChain;
+import com.microsoft.azure.functions.internal.spi.middleware.MiddlewareContext;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.ManagedContext;
+
+public class QuarkusAzureFunctionsMiddleware implements Middleware {
+    @Override
+    public void invoke(MiddlewareContext middlewareContext, MiddlewareChain middlewareChain) throws Exception {
+        ManagedContext requestContext = Arc.container().requestContext();
+        requestContext.activate();
```

Comment:
You probably want to check if the context is already active

Reply:
Its not possible for request context to be active.  This "interceptor" is driven/called by Azure Functions runtime when a new request comes in only.  They don't have "local" invocations.

Reply:
Fair enough